diff --git a/host/drivers/bt/vhci_drv.c b/host/drivers/bt/vhci_drv.c
index 9e172a7..b615df0 100644
--- a/host/drivers/bt/vhci_drv.c
+++ b/host/drivers/bt/vhci_drv.c
@@ -29,6 +29,10 @@ static const char TAG[] = "vhci_drv";
 #define BLE_HCI_CMD_HDR_LEN                 (3)
 #endif
 
+#ifndef __WEAK__
+#define __WEAK__	__attribute__ ((weak))
+#endif
+
 void hci_drv_init(void)
 {
 	// do nothing for VHCI: underlying transport should be ready
